src/netif.rs

```diff
@@ -1102,14 +1207,16 @@ mod driver {
         ) -> i32 {
             let this = unsafe { (h as *mut Self).as_mut() }.unwrap();
             let data = core::slice::from_raw_parts(buffer as *mut u8, len);
+
+            #[allow(clippy::let_and_return)]
             let result = match this.tx(data) {
                 Ok(_) => ESP_OK,
                 Err(e) => e.code(),
             };
 
             // TODO: Might not be necessary, but if I remember correctly, the Netif API
             // wanted that _we_ free the buffer; in any case needs to be compared with the C ESP Modem code
-            free(buffer);
+            // free(buffer);
```

Comment:
did it turned out to not be necessary ?

Reply:
No idea for now :) that's why I've kept the comment for now. Hopefully, once the esp modem is relatively complete, we'll know...